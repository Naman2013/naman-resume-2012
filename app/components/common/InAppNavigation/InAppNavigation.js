import React from 'react';
import PropTypes from 'prop-types';
import style from './InAppNavigation.style';
import { horizontalArrow } from 'styles/variables/iconURLs';

const InAppNavigation = ({ previousText, title }) => (
  <div className="root">
    <div className="back-text-container">
      <img className="icon-arrow" alt="" src={horizontalArrow} />
      <p className="back-text">back{previousText && ` to ${previousText}`}</p>
    </div>
    <h5 className="title">{title}</h5>
    <style jsx>{style}</style>
  </div>
);

InAppNavigation.propTypes = {
  previousText: PropTypes.string,
  title: PropTypes.string.isRequired,
};

InAppNavigation.defaultProps = {
  previousText: '',
};

export default InAppNavigation;
