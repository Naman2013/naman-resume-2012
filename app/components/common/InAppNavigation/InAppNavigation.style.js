import css from 'styled-jsx/css';
import { primaryFont, secondaryFont } from 'styles/variables/fonts';
import { astronaut } from 'styles/variables/colors_tiles_v4';
import { screenMedium } from 'styles/variables/breakpoints';

export default css`
  .back-text-container {
    padding: 20px 25px;
    display: flex;
    align-items: center;
  }

  .icon-arrow {
    transform: rotate(180deg);
  }

  .back-text {
    padding: 0;
    margin: 0;
    letter-spacing: 1.25px;
    font-family: ${primaryFont};
    text-transform: uppercase;
    background-color: white;
    padding-left: 12px;
    font-size: 12px;
    font-weight: 800;
    color: ${astronaut};
  }

  .title {
    padding: 40px;
    margin: 0;
    font-family: ${secondaryFont};
    background-color: #253446;
    color: white;
    font-weight: 100;
    text-transform: capitalize;
    font-size: 16px;
  }
`;
