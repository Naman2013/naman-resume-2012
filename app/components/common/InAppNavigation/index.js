import InAppNavigation from './InAppNavigation';

export default InAppNavigation;
