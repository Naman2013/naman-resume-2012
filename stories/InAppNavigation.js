import React from 'react';
import { storiesOf } from '@storybook/react';
import InAppNavigation from '../app/components/common/InAppNavigation';

storiesOf('InAppNavigation', module)
  .add('With `previousText`', () => (
    <InAppNavigation title="Topic: 1 Astronomical Time" previousText="guide" />
  ))
  .add('Without `previousText`', () => (
    <InAppNavigation title="Topic: 1 Astronomical Time" />
  ));
