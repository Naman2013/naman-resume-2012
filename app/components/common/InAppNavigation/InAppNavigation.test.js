import React from 'react';
import { shallow } from 'enzyme';
import InAppNavigation from './InAppNavigation';

describe('InAppNavigation', () => {
  const shallowWrapper = shallow(<InAppNavigation />);
  it('should render correctly', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('should render previous page name when provided', () => {
    shallowWrapper.setProps({ previousText: 'guide' });
    expect(shallowWrapper.find('.back-text').text()).toBe('back to guide');
  });

  it('should render back as the text in the back button when no back page is provided', () => {
    shallowWrapper.setProps({ previousText: '' });
    expect(shallowWrapper.find('.back-text').text()).toBe('back');
  });
});
