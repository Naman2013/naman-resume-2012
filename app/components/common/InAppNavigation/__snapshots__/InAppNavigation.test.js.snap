// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InAppNavigation should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InAppNavigation
    previousText="guide"
    title="topic 1: stuff..."
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="jsx-1574755409 back-text-container"
        >
          <img
            alt=""
            className="jsx-1574755409 icon-arrow"
            src="https://vega.slooh.com/assets/v4/common/arrow_horz.svg"
          />
          <p
            className="jsx-1574755409 back-text"
          >
            back
             to guide
          </p>
        </div>,
        <h5
          className="jsx-1574755409 title"
        >
          topic 1: stuff...
        </h5>,
        <JSXStyle
          css={
            Array [
              ".back-text-container.jsx-1574755409{padding:20px 25px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
              ".icon-arrow.jsx-1574755409{-webkit-transform:rotate(180deg);-ms-transform:rotate(180deg);transform:rotate(180deg);}",
              ".back-text.jsx-1574755409{padding:0;margin:0;-webkit-letter-spacing:1.25px;-moz-letter-spacing:1.25px;-ms-letter-spacing:1.25px;letter-spacing:1.25px;font-family:brandon-grotesque, sans-serif;text-transform:uppercase;background-color:white;padding-left:12px;font-size:12px;font-weight:800;color:#415671;}",
              ".title.jsx-1574755409{padding:40px;margin:0;font-family:adobe-garamond-pro, serif;background-color:#253446;color:white;font-weight:100;text-transform:capitalize;font-size:16px;}",
            ]
          }
          styleId="1574755409"
        />,
      ],
      "className": "jsx-1574755409 root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt=""
              className="jsx-1574755409 icon-arrow"
              src="https://vega.slooh.com/assets/v4/common/arrow_horz.svg"
            />,
            <p
              className="jsx-1574755409 back-text"
            >
              back
               to guide
            </p>,
          ],
          "className": "jsx-1574755409 back-text-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "className": "jsx-1574755409 icon-arrow",
              "src": "https://vega.slooh.com/assets/v4/common/arrow_horz.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "back",
                " to guide",
              ],
              "className": "jsx-1574755409 back-text",
            },
            "ref": null,
            "rendered": Array [
              "back",
              " to guide",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "topic 1: stuff...",
          "className": "jsx-1574755409 title",
        },
        "ref": null,
        "rendered": "topic 1: stuff...",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [
            ".back-text-container.jsx-1574755409{padding:20px 25px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
            ".icon-arrow.jsx-1574755409{-webkit-transform:rotate(180deg);-ms-transform:rotate(180deg);transform:rotate(180deg);}",
            ".back-text.jsx-1574755409{padding:0;margin:0;-webkit-letter-spacing:1.25px;-moz-letter-spacing:1.25px;-ms-letter-spacing:1.25px;letter-spacing:1.25px;font-family:brandon-grotesque, sans-serif;text-transform:uppercase;background-color:white;padding-left:12px;font-size:12px;font-weight:800;color:#415671;}",
            ".title.jsx-1574755409{padding:40px;margin:0;font-family:adobe-garamond-pro, serif;background-color:#253446;color:white;font-weight:100;text-transform:capitalize;font-size:16px;}",
          ],
          "styleId": "1574755409",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="jsx-1574755409 back-text-container"
          >
            <img
              alt=""
              className="jsx-1574755409 icon-arrow"
              src="https://vega.slooh.com/assets/v4/common/arrow_horz.svg"
            />
            <p
              className="jsx-1574755409 back-text"
            >
              back
               to guide
            </p>
          </div>,
          <h5
            className="jsx-1574755409 title"
          >
            topic 1: stuff...
          </h5>,
          <JSXStyle
            css={
              Array [
                ".back-text-container.jsx-1574755409{padding:20px 25px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
                ".icon-arrow.jsx-1574755409{-webkit-transform:rotate(180deg);-ms-transform:rotate(180deg);transform:rotate(180deg);}",
                ".back-text.jsx-1574755409{padding:0;margin:0;-webkit-letter-spacing:1.25px;-moz-letter-spacing:1.25px;-ms-letter-spacing:1.25px;letter-spacing:1.25px;font-family:brandon-grotesque, sans-serif;text-transform:uppercase;background-color:white;padding-left:12px;font-size:12px;font-weight:800;color:#415671;}",
                ".title.jsx-1574755409{padding:40px;margin:0;font-family:adobe-garamond-pro, serif;background-color:#253446;color:white;font-weight:100;text-transform:capitalize;font-size:16px;}",
              ]
            }
            styleId="1574755409"
          />,
        ],
        "className": "jsx-1574755409 root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt=""
                className="jsx-1574755409 icon-arrow"
                src="https://vega.slooh.com/assets/v4/common/arrow_horz.svg"
              />,
              <p
                className="jsx-1574755409 back-text"
              >
                back
                 to guide
              </p>,
            ],
            "className": "jsx-1574755409 back-text-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "className": "jsx-1574755409 icon-arrow",
                "src": "https://vega.slooh.com/assets/v4/common/arrow_horz.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "back",
                  " to guide",
                ],
                "className": "jsx-1574755409 back-text",
              },
              "ref": null,
              "rendered": Array [
                "back",
                " to guide",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "topic 1: stuff...",
            "className": "jsx-1574755409 title",
          },
          "ref": null,
          "rendered": "topic 1: stuff...",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [
              ".back-text-container.jsx-1574755409{padding:20px 25px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
              ".icon-arrow.jsx-1574755409{-webkit-transform:rotate(180deg);-ms-transform:rotate(180deg);transform:rotate(180deg);}",
              ".back-text.jsx-1574755409{padding:0;margin:0;-webkit-letter-spacing:1.25px;-moz-letter-spacing:1.25px;-ms-letter-spacing:1.25px;letter-spacing:1.25px;font-family:brandon-grotesque, sans-serif;text-transform:uppercase;background-color:white;padding-left:12px;font-size:12px;font-weight:800;color:#415671;}",
              ".title.jsx-1574755409{padding:40px;margin:0;font-family:adobe-garamond-pro, serif;background-color:#253446;color:white;font-weight:100;text-transform:capitalize;font-size:16px;}",
            ],
            "styleId": "1574755409",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
